src/browser/api/browser_view.ts

```diff
@@ -89,12 +100,26 @@ export async function attach(ofView: OfView, toIdentity: Identity) {
         }
         windowCloseListenerMap.get(ofWin).set(ofView, listener);
         updateViewTarget(ofView, toIdentity);
+        of_events.emit(route.view('attached', ofView.uuid, ofView.name), {
+            name: ofView.name,
+            uuid: ofView.uuid,
+            target: toIdentity,
+            previousTarget
+        });
+        of_events.emit(route.view('shown', ofView.uuid, ofView.name), {
+            name: ofView.name,
+            uuid: ofView.uuid,
+            target: toIdentity
+        });
     }
 }
 function destroy (ofView: OfView) {
-   removeBrowserView(ofView);
-   ofView.view.destroy();
+    const {uuid, name, target, view} = ofView;
+    removeBrowserView(ofView);
+    view.destroy();
+    of_events.emit(route.view('destroyed', uuid, name), {name, uuid, target});
```

Comment:
If browserview creation fails this could cause an event for a view which doesn't exist

Reply:
Good call, let's pow-wow and figure out how we want this to act.